store/mockstore/unistore/cophandler/mpp_exec.go

```diff
@@ -0,0 +1,572 @@
+	length := len(e.groupByExprs)
+	if length == 0 {
+		return nil, nil
+	}
+	key := make([]byte, 0, 32)
+	for _, item := range e.groupByExprs {
+		v, err := item.Eval(row)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		b, err := codec.EncodeValue(e.sc, nil, v)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		key = append(key, b...)
+	}
+	return key, nil
+}
+
+func (e *aggExec) getContexts(groupKey []byte) []*aggregation.AggEvaluateContext {
+	aggCtxs, ok := e.aggCtxsMap[string(groupKey)]
+	if !ok {
+		aggCtxs = make([]*aggregation.AggEvaluateContext, 0, len(e.aggExprs))
+		for _, agg := range e.aggExprs {
+			aggCtxs = append(aggCtxs, agg.CreateContext(e.sc))
+		}
+		e.aggCtxsMap[string(groupKey)] = aggCtxs
+	}
+	return aggCtxs
+}
+
+func (e *aggExec) processAllRows() (*chunk.Chunk, error) {
+	for {
+		chk, err := e.children[0].next()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if chk == nil {
+			break
+		}
+		rows := chk.NumRows()
+		i := 0
+		for i < rows {
+			row := chk.GetRow(i)
+			i++
+			gk, err := e.getGroupKey(row)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			if _, ok := e.groups[string(gk)]; !ok {
+				e.groups[string(gk)] = struct{}{}
+				e.groupKeys = append(e.groupKeys, gk)
+			}
+
+			aggCtxs := e.getContexts(gk)
+			for i, agg := range e.aggExprs {
+				err = agg.Update(aggCtxs[i], e.sc, row)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+			}
+		}
+	}
+
+	chk := chunk.NewChunkWithCapacity(e.fieldTypes, 0)
+
+	for _, gk := range e.groupKeys {
+		newRow := chunk.MutRowFromTypes(e.fieldTypes)
+		aggCtxs := e.getContexts(gk)
+		for i, agg := range e.aggExprs {
+			partialResults := agg.GetPartialResult(aggCtxs[i])
+			newRow.SetDatum(i, partialResults[0])
+		}
+		chk.AppendRow(newRow.ToRow())
+	}
+	return chk, nil
+}
+
+func (e *aggExec) next() (*chunk.Chunk, error) {
+	if !e.processed {
+		e.processed = true
+		return e.processAllRows()
+	}
+	return nil, nil
+}
+
+type selExec struct {
+	baseMPPExec
+
+	conditions []expression.Expression
+}
+
+func (e *selExec) open() error {
+	return e.children[0].open()
+}
+
+func (e *selExec) next() (*chunk.Chunk, error) {
+	chk, err := e.children[0].next()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if chk == nil {
+		return nil, nil
+	}
+	for rows := chk.NumRows() - 1; rows >= 0; rows-- {
+		row := chk.GetRow(rows)
+		for _, cond := range e.conditions {
+			d, err := cond.Eval(row)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+
+			var passCheck bool
+			if d.IsNull() {
+				passCheck = false
+			} else {
+				isBool, err := d.ToBool(e.sc)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				isBool, err = expression.HandleOverflowOnSelection(e.sc, isBool, err)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				passCheck = isBool != 0
+			}
+			if !passCheck {
+				chk.TruncateTo(rows)
+				break
+			}
```

Comment:
what if only the first row of a chunk does not pass the check?

Reply:
This loop is for conditions instead of rows